src/timestamp9.c

```diff
@@ -647,3 +650,26 @@ Datum timestamp9_interval_mi(PG_FUNCTION_ARGS)
 	new_ts += ts % 1000;
 	PG_RETURN_TIMESTAMP9(new_ts);
 }
+
+Datum
+timestamp9_nsnow(PG_FUNCTION_ARGS)
+{
+	timestamp9 result;
+	struct timespec tv;
+	static TimestampTz current_xact_timestamp;
```

Comment:
Does this work in the case parallelism is enabled in queries? You can try using `force_parallel_mode` to test.

Reply:
Nice catch! It seems that the `current_xact_ts9_timestamp` cannot be synced to parallel workers in my implementation like `xactStartTimestamp`. Maybe we can mark it with `PARALLEL RESTRICTED`?

https://www.postgresql.org/docs/current/parallel-safety.html